test/pangolin-exchange.test.ts

```diff
@@ -0,0 +1,104 @@
+
+const {deployContract, provider} = waffle;
+const {expect} = chai;
+
+const pangolinRouterAddress = '0xE54Ca86531e17Ef3616d22Ca28b0D458b6C89106';
+const daiETokenAddress = '0xd586E7F844cEa2F87f50152665BCbc2C279D8d70';
+const WAVAXTokenAddress = '0xB31f66AA3C1e785363F0875A1B74E27b85FD66c7';
+
+const ERC20Abi = [
+  'function decimals() public view returns (uint8)',
+  'function balanceOf(address _owner) public view returns (uint256 balance)',
+  'function approve(address _spender, uint256 _value) public returns (bool success)'
+]
+
+const pangolinRouterAbi = [
+  'function getAmountsIn (uint256 amountOut, address[] path) view returns (uint256[])'
+]
+
+
+describe('PangolinExchange', () => {
+  describe('Test buying and selling an asset', () => {
+    let sut: PangolinExchange,
+      daie: Contract,
+      pgrt: Contract,
+      owner: SignerWithAddress;
+
+    before('Deploy the PangolinExchange contract', async () => {
+      [owner] = await getFixedGasSigners(10000000);
+      sut = await deployContract(owner, PangolinExchangeArtifact, [pangolinRouterAddress]) as PangolinExchange;
+      daie = await new ethers.Contract(daiETokenAddress, ERC20Abi);
+      pgrt = await new ethers.Contract(pangolinRouterAddress, pangolinRouterAbi);
+    });
+
+
+    it('should check for the amount of tokens to buy to be greater than 0', async () => {
+      await expect(sut.buyERC20Token(daiETokenAddress, 0)).to.be.revertedWith('Amount of tokens to buy has to be greater than 0');
+    });
+
+
+    it('should check if enough funds were provided', async () => {
+      const daiePurchaseAmount = 1e18;
+      const estimatedAvax = (await pgrt.connect(owner).getAmountsIn(daiePurchaseAmount.toString(), [WAVAXTokenAddress, daiETokenAddress]))[0];
+
+      await expect(sut.buyERC20Token(daiETokenAddress, daiePurchaseAmount.toString(), {value: Math.floor(estimatedAvax*0.9).toString()})).to.be.revertedWith('Not enough funds provided');
+    });
+
+
+    it('should check if an erc20 tokens were purchased successfully', async () => {
+      const DAIePurchaseAmount = 1e18;
+      const estimatedAvax = (await pgrt.connect(owner).getAmountsIn(DAIePurchaseAmount.toString(), [WAVAXTokenAddress, daiETokenAddress]))[0];
+      const initialDAIeBalance = await daie.connect(owner).balanceOf(owner.address);
+      const initialAvaxBalance = await provider.getBalance(owner.address);
+
+      expect(initialDAIeBalance).to.equal(0);
+
+      await sut.buyERC20Token(daiETokenAddress, DAIePurchaseAmount.toString(), {value: estimatedAvax.toString()});
+
+      const currentDAIeBalance = await daie.connect(owner).balanceOf(owner.address);
+      const currentAvaxBalance = await provider.getBalance(owner.address);
+      const avaxBalanceDifference = BigNumber.from(initialAvaxBalance.toString()).sub(BigNumber.from(estimatedAvax.toString()));
+
+      expect(currentDAIeBalance).to.equal(DAIePurchaseAmount.toString());
+      expect(currentAvaxBalance).to.be.lt(avaxBalanceDifference);
```

Comment:
According to the uniswapv2 documentation (https://docs.uniswap.org/protocol/V2/reference/smart-contracts/library#getamountsin) the estimatedAvax will be equal to the minimum amount of AVAX we have to pay in order to purchase daiTokenPurchaseAmount of DAI tokens.

So we expect the avax balance after buying the DAI tokens to be equal to or less than initialAvaxBalance-estimatedAvax . I will change the condition to lte.

I also changed the avaxBalanceDifference  name to expectedAvaxBalance so that it is more descriptive.

Does the above make sense or maybe I am mistaken?